src/webapi/Service/PeriodicHostedSyncOdsAssetsService.cs

```diff
@@ -0,0 +1,57 @@
+namespace eppeta.webapi.Service
+{
+    class PeriodicHostedSyncOdsAssetsService : BackgroundService
+    {
+        private readonly TimeSpan _period = TimeSpan.FromHours(AppSettings.SyncOdsAssetsSettings.PeriodInHours);
+        private readonly ILogger<PeriodicHostedSyncOdsAssetsService> _logger;
+        private readonly IServiceScopeFactory _factory;
+        private long _executionCount = 0;
+        public bool IsEnabled { get; set; } = true;
+
+        public PeriodicHostedSyncOdsAssetsService(ILogger<PeriodicHostedSyncOdsAssetsService> logger, IServiceScopeFactory factory)
+        {
+            _logger = logger;
+            _factory = factory;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            // ExecuteAsync is executed once and we have to take care of a mechanism ourselves that is kept during operation.
+            // To do this, we can use a Periodic Timer, which, unlike other timers, does not block resources.
+            // But instead, WaitForNextTickAsync provides a mechanism that blocks a task and can thus be used in a While loop.
+            using PeriodicTimer timer = new PeriodicTimer(_period);
+
+            // When ASP.NET Core is intentionally shut down, the background service receives information
+            // via the stopping token that it has been canceled.
+            // We check the cancellation to avoid blocking the application shutdown.
+            while (
+                !stoppingToken.IsCancellationRequested &&
+                await timer.WaitForNextTickAsync(stoppingToken))
```

Comment:
@DavidJGapCR I figured out the issue I was having getting this to sync. `await timer.WaitForNextTickAsync(stoppingToken))` is only true after `_period` has elapsed the first time, this means that from initial startup of the WebApi it will take a minimum of an hour before sync will run. 

In the field I would expect sync to be scheduled every 24 hours or so, but the initial load of data might be a problem because of how long it will take to execute. I'm wondering about adding another endpoint that would allow us to manually sync, this would allow us to keep the existing code and would also allow an administrator to sync as they see fit (we would add a page for this is the future) Let's discuss on standup in the morning

Reply:
That is a good point. I should have thought of that. I'll try to think of alternatives.